### PR TITLE
Adding basic docs for read only tokens.

### PR DESCRIPTION
**Pull request by logan-anderson**
Starting work do be done in https://github.com/tinacms/tina-cloud/issues/1398

# Update:

Ready for review but can not be merged until we release read only tokens.

### CODE REVIEW

**Comment by logan-anderson** [comment]
Can add more context once https://github.com/tinacms/tina-cloud/pull/1395 is merged and I am able to test out the generation of the tokens.

**Comment by logan-anderson** [comment]
closes https://github.com/tinacms/tina-cloud/issues/1404

File: content/docs/graphql/read-only-tokens.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
+---
+title: Read Only Tokens
+last_edited: '2022-02-07T18:00:00.000Z'
+---
+{{ WarningCallout text="This is an experimental feature and may be slow as we work on performance improvements" }}
+
+Read only tokens allow data fetching at runtime without the need for the local graphQL server. Some use cases might include the following.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:7]:
```suggestion
Read-only tokens allow data fetching at runtime without the need for the local graphQL server. Some use cases include the following:
```

File: content/docs/graphql/read-only-tokens.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+
+- Runtime server side logic in `getServerSideProps`, `getStaticProps` (when fallback is not `false`), etc.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:10]:
```suggestion
- Runtime server-side logic in `getServerSideProps`, `getStaticProps` (when fallback is not `false`), etc.
```

File: content/docs/graphql/read-only-tokens.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+- [Incremental Static Site Generation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration)
+- [Server components](https://nextjs.org/docs/advanced-features/react-18#react-server-components)
+- [Next.js middleware](https://nextjs.org/docs/middleware)
+- Client side data fetching (including `create-react-app`)
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:14]:
```suggestion
- Client-side data-fetching (including `create-react-app`)
```

File: content/docs/graphql/read-only-tokens.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+- Future support for server side frameworks like [remix](https://remix.run/)
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:15]:
```suggestion
- Future support for server-side frameworks like [remix](https://remix.run/)
```

File: content/docs/graphql/read-only-tokens.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+In all of these use cases we can no longer rely static content but need a way to fetch data in real-time without being authenticated.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:17]:
```suggestion
In all of these use cases we can no longer rely on static content, but need a way to fetch data in real-time without being authenticated.
```

File: content/docs/graphql/read-only-tokens.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+
+Read only tokens can only be used for [GraphQL query requests]() and can not be used for mutations. It can also only be used to get data from the branch in the list of branches specified when the token was made.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:20]:
```suggestion
Read-only tokens can only be used for [GraphQL query requests]() and can not be used for mutations. They can also only be used to get data from the branch in the list of branches specified when the token was made. 
```

File: content/docs/graphql/read-only-tokens.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+Read only tokens can only be used for [GraphQL query requests]() and can not be used for mutations. It can also only be used to get data from the branch in the list of branches specified when the token was made. 
+
+## How to use Read Only Tokens
+
+### Generate them from the dashboard
+
+Navigate to [Tina Cloud](https://app.tina.io) and click on the project you wish to add a token to, click on the "tokens" tab
+![Tina cloud token tab](/img/graphql-docs/token-tab.png)
+
+Next, click "New Token" and fill out fields. The token name is how you can identify the token and "Git branches" is the list of branches separated by commas that the token has assess too.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:29]:
```suggestion
Next, click "New Token" and fill out fields. The token name is how you can identify the token and "Git branches" is the list of branches separated by commas that the token has access too. 
```

File: content/docs/graphql/read-only-tokens.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+Finally, click "Create Token".
+
+![Successful creation of a token in Tina Cloud](/img/graphql-docs/final-token-page.png)
+
+
+### Make a fetch request using the API key 
+
+Now you can make a POST request to the content API with the desired GraphQL request.
+
+The endpoint is `https://content.tinajs.io/content/<myClientId>/github/<myBranch>` and the token can be passed by including a `X-API-KEY` with the token as the value.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:43]:
```suggestion
The endpoint is `https://content.tinajs.io/content/<myClientId>/github/<myBranch>` and the token can be passed by including the`X-API-KEY` header with the token as the value.
```

File: content/docs/graphql/read-only-tokens.md [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+
+Here is an example curl command that will query the content API for the list of collections.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:45]:
```suggestion
Here is an example curl request that will query the content API for the list of collections:
```

File: content/docs/graphql/read-only-tokens.md [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+};
+
+fetch("https://content.tinajs.io/content/<ClientId>/github/main", requestOptions)
+  .then(response => response.text())
+  .then(result => console.log(result))
+  .catch(error => console.log('error', error));
+```
+
+
+## Examples of using read only tokens
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:78]:
```suggestion
## Examples of using read-only tokens
```

File: content/docs/graphql/read-only-tokens.md [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+
+### Fetch data client side
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:80]:
```suggestion
### Fetch data client-side
```

File: content/docs/graphql/read-only-tokens.md [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+      <p>{data.body}</p>
+    </div>
+  );
+}
+export default BlogPostPage;
+```
+
+### Next.js `fallback: "blocking"`
+
+In Next.js one can specify [`fallback: "blocking"`](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking), this allows `getStaticProps` to run server side at runtime when a user goes to a page that was not specified in `getStaticPaths`.
```
Comment (kldavis4) [inline content/docs/graphql/read-only-tokens.md:146]:
```suggestion
In Next.js one can specify [`fallback: "blocking"`](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking), this allows `getStaticProps` to run server-side at request time when a user goes to a page that was not specified in `getStaticPaths`. 
```

**Review by kldavis4 — COMMENTED** [review]
minor editing cleanup

**Comment by logan-anderson** [comment]
Thank you @kldavis4 😄

File: content/docs/graphql/read-only-tokens.md [hunk 13]
```diff
@@ -0,0 +1,217 @@
+---
+title: Read-only Tokens
+last_edited: '2022-02-07T18:00:00.000Z'
+---
+{{ WarningCallout text="This is an experimental feature and may be slow as we work on performance improvements" }}
+
+Read-only tokens allow data fetching at runtime without the need for the local graphQL server. Some use cases include the following:
```
Comment (jhuggett) [inline content/docs/graphql/read-only-tokens.md:7]:
```suggestion
Read-only tokens allow data fetching at runtime without the need for the local GraphQL server. Some use cases include the following:
```

File: content/docs/graphql/read-only-tokens.md [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+
+
+- Runtime server-side logic in `getServerSideProps`, `getStaticProps` (when fallback is not `false`), etc.
+- [Incremental Static Site Generation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration)
+- [Server components](https://nextjs.org/docs/advanced-features/react-18#react-server-components)
```
Comment (jhuggett) [inline content/docs/graphql/read-only-tokens.md:12]:
```suggestion
- [Server components](https://nextjs.org/docs/advanced-features/react-18/overview#react-server-components-alpha)
```
seems to link to the server components heading, the other just goes to the same page but at the top

**Pull request merged**